src/components/Header/Header.js

```diff
@@ -12,9 +11,8 @@ import {
   NotificationNew24,
 } from '@carbon/icons-react';
 import { settings } from 'carbon-components';
+import FocusTrap from 'focus-trap-react';
```

Comment:
I made a few changes here to manage:

1. Focus traps within the different menus and the sidenav
2. Handling the "ESC" key to exit out of the sidenav and menus and returning focus to the button associated w/ the menu.
3. Tab to leave the side panel (not sure if this is the best approach but its not used afaik at the moment)
4. Removed dead code.